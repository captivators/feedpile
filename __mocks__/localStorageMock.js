const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"faiz"}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
