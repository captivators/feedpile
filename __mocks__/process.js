const process = {
  "env" : {
    "AUTH_ID":"test",
    "AUTH_DOMAIN":"user_id"
  }
}
global.process = process;
