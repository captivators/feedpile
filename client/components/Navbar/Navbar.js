import React from 'react';
import './Navbar.css';

const Navbar = () => {
  return (
    <div className="navbar">
      <div className="nav-item-container">
        <p className="logout">Logout</p>
      </div>
    </div>
  )
}

export default Navbar;
