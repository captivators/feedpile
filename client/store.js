import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
