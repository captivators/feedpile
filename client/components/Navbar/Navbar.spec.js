import '../../../__mocks__/localStorageMock'
import React from 'react';
import FlatButton from 'material-ui/FlatButton';
import {Provider} from 'react-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import store from '../../store';
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Navbar from './Navbar'



test('Navbar take a snapshot', () => {
  const component = shallow(<Navbar store={ store }/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

// test('Navbar renders a greeting and an Avatar', () => {
//   // const component = render(<MuiThemeProvider><Navbar store={ store }/></MuiThemeProvider>)
//   // expect(component.find('.greeting').length).toEqual(1)
//
//
//   const component = mount(<MuiThemeProvider><Navbar store={ store }/></MuiThemeProvider>)
//   const element = component.find('.navbar');
//   expect(element.length).toEqual(1);
//
// // Use the Node property of the Wrapped Element to expose the component instance
//   const element2 = component.find(FlatButton);
//   console.log(element2.debug())
//   // const node = ReactDOM.findDOMNode(element2.node)
//   element2.simulate('click');
//
//   expect(element2.state().selected).to.be.true
// })