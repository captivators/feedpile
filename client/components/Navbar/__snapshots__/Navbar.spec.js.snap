// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar take a snapshot 1`] = `
<Navbar
  logoutSuccess={[Function]}
  profile={
    Object {
      "identities": Array [
        Object {
          "user_id": 123,
        },
      ],
      "name": "test",
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
